### ISSUE TITLE
getMarkRange should be handled differently for link

### ISSUE THREAD

**Issue by hasakilol** [ueberdosis/tiptap#3872]
### What’s the bug you are facing?

If two links are ajacent with no white space between, getMarkRange will return the range containing both the links, but not one.

### Which browser was this experienced in? Are any special extensions installed?

Chrome

### How can we reproduce the bug on our side?

I will provide with a demo later. Please don't delete my post.

### What did you expect to happen?

Range with only one link is returned.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by hasakilol** [comment]
```
let startIndex = start.index
  let startPos = $pos.start() + start.offset
  let endIndex = startIndex + 1
  let endPos = startPos + start.node.nodeSize

  findMarkInSet([...start.node.marks], type, attributes) **// this code seems useless???**

  while (startIndex > 0 && mark.isInSet($pos.parent.child(startIndex - 1).marks)) {
    startIndex -= 1
    startPos -= $pos.parent.child(startIndex).nodeSize
  }

  while (
    endIndex < $pos.parent.childCount
    && isMarkInSet([...$pos.parent.child(endIndex).marks], type, attributes) 
    **// in git history, this was mark.isInSet($pos.parent.child(endIndex).marks). Why did the change happen?**
  ) {
    endPos += $pos.parent.child(endIndex).nodeSize
    endIndex += 1
  }
```

**Comment by nperez0111** [comment]
This is released with 2.10.0
